src/js/integrations/mobilereplay.ts

```diff
@@ -0,0 +1,129 @@
+import type { Client, DynamicSamplingContext, Event, IntegrationFnResult } from '@sentry/types';
+import { logger } from '@sentry/utils';
+
+import { isHardCrash } from '../misc';
+import { hasHooks } from '../utils/clientutils';
+import { isExpoGo, notMobileOs } from '../utils/environment';
+import { NATIVE } from '../wrapper';
+
+export const MOBILE_REPLAY_INTEGRATION_NAME = 'MobileReplay';
+
+export interface MobileReplayOptions {
+  /**
+   * Mask all text in recordings
+   */
+  maskAllText?: boolean;
+
+  /**
+   * Mask all text in recordings
+   */
+  maskAllImages?: boolean;
+}
+
+const defaultOptions: Required<MobileReplayOptions> = {
+  maskAllText: true,
+  maskAllImages: true,
+};
+
+type MobileReplayIntegration = IntegrationFnResult & {
+  options: Required<MobileReplayOptions>;
+};
+
+/**
+ * The Mobile Replay Integration, let's you adjust the default mobile replay options.
+ * To be passed to `Sentry.init` with `replaysOnErrorSampleRate` or `replaysSessionSampleRate`.
+ *
+ * ```javascript
+ * Sentry.init({
+ *  _experiments: {
+ *    replaysOnErrorSampleRate: 1.0,
+ *    replaysSessionSampleRate: 1.0,
+ *  },
+ *  integrations: [mobileReplayIntegration({
+ *    // Adjust the default options
+ *  })],
+ * });
+ * ```
+ *
+ * @experimental
+ */
+export const mobileReplayIntegration = (initOptions: MobileReplayOptions = defaultOptions): MobileReplayIntegration => {
+  if (isExpoGo()) {
+    logger.warn(
+      `[Sentry] ${MOBILE_REPLAY_INTEGRATION_NAME} is not supported in Expo Go. Use EAS Build or \`expo prebuild\` to enable it.`,
+    );
+  }
+  if (notMobileOs()) {
+    logger.warn(`[Sentry] ${MOBILE_REPLAY_INTEGRATION_NAME} is not supported on this platform.`);
+  }
+
+  if (isExpoGo() || notMobileOs()) {
+    return mobileReplayIntegrationNoop();
+  }
+
+  const options = { ...defaultOptions, ...initOptions };
+
+  async function processEvent(event: Event): Promise<Event> {
+    const hasException = event.exception && event.exception.values && event.exception.values.length > 0;
+    if (!hasException) {
+      // Event is not an error, will not capture replay
+      return event;
+    }
+
+    const recordingReplayId = NATIVE.getCurrentReplayId();
+    if (recordingReplayId) {
+      logger.debug(
+        `[Sentry] ${MOBILE_REPLAY_INTEGRATION_NAME} assign already recording replay ${recordingReplayId} for event ${event.event_id}.`,
+      );
+      return event;
+    }
+
+    const replayId = await NATIVE.startReplay(isHardCrash(event));
```

Comment:
It was quite confusing to me that this is called `startReplay` here but then actually calls into captureReplay in native to (pressumably) capture a running replay buffer. WDYT about renaming to captureReplay instead?

Reply:
Yes, I absolutely agree.